internal/cmd/stack/environment.go

```diff
@@ -0,0 +1,157 @@
+package stack
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/shurcooL/graphql"
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+// ConfigType is a type of configuration element.
+type ConfigType string
+
+// ConfigInput represents the input required to create or update a config
+// element.
+type ConfigInput struct {
+	ID        graphql.ID      `json:"id"`
+	Type      ConfigType      `json:"type"`
+	Value     graphql.String  `json:"value"`
+	WriteOnly graphql.Boolean `json:"writeOnly"`
+}
+
+func setVar(cliCtx *cli.Context) error {
+	if nArgs := cliCtx.NArg(); nArgs != 2 {
+		return fmt.Errorf("expecting environment setenv as two arguments, got %d instead", nArgs)
+	}
+
+	envName := cliCtx.Args().Get(0)
+	envValue := cliCtx.Args().Get(1)
+
+	stackID := cliCtx.String(flagStackID.Name)
+
+	var mutation struct {
+		ConfigElement struct {
+			ID        string `graphql:"id"`
+			WriteOnly bool   `graphql:"writeOnly"`
+			Value     string `graphql:"value"`
+		} `graphql:"stackConfigAdd(stack: $stack, config: $config)"`
+	}
+
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+		"config": ConfigInput{
+			ID:        graphql.ID(envName),
+			Type:      ConfigType("ENVIRONMENT_VARIABLE"),
+			Value:     graphql.String(envValue),
+			WriteOnly: graphql.Boolean(cliCtx.Bool(flagEnvironmentWriteOnly.Name)),
+		},
+	}
+
+	if err := authenticated.Client.Mutate(context.Background(), &mutation, variables); err != nil {
```

Comment:
You can use `cliCtx.Context` here. Not sure if it would make any difference, but figure if there's already a context available we might as well use it.